analyzers/src/SonarAnalyzer.Common/Rules/InsteadOfAnyBase.cs

```diff
@@ -62,7 +69,8 @@ protected InsteadOfAnyBase()
                 && Language.Syntax.HasExactlyNArguments(invocation, 1)
                 && Language.Syntax.TryGetOperands(invocation, out var left, out var right)
                 && IsCorrectCall(right, c.SemanticModel)
-                && c.SemanticModel.GetTypeInfo(left).Type is { } type)
+                && c.SemanticModel.GetTypeInfo(left).Type is { } type
+                && !IsEntityFramework(c.Node, c.SemanticModel))
```

Comment:
instead of `c.Node` you could use `invocation` - it's a bit confusing otherwise IMO